src/System.Private.CoreLib/shared/System/Lazy.cs

```diff
@@ -351,7 +351,7 @@ private void ExecutionAndPublication(LazyHelper executionAndPublication, bool us
 
         private void PublicationOnly(LazyHelper publicationOnly, T possibleValue)
         {
-            LazyHelper previous = Interlocked.CompareExchange(ref _state, LazyHelper.PublicationOnlyWaitForOtherThreadToPublish, publicationOnly);
+            LazyHelper? previous = Interlocked.CompareExchange(ref _state, LazyHelper.PublicationOnlyWaitForOtherThreadToPublish, publicationOnly);
```

Comment:
Would it be more efficient if it was `Interlocked.CompareExchange<LazyHelper>(ref _state, LazyHelper.PublicationOnlyWaitForOtherThreadToPublish, publicationOnly)` ?

Reply:
Or maybe that doesn't work because you want `LazyHelper?`

Reply:
More efficient?

Reply:
I guess I figured it must be using `public static object CompareExchange (ref object location1, object value, object comparand);` and on the generic version is noted `This method overload is preferable to the CompareExchange(Object, Object, Object) method overload, because the latter requires the destination object to be accessed late-bound.` (whatever that means)

But now I see the compiler prefers the generic overload unless the parameter is explicitly of type object.